test/commons/text/accessible-text.js

```diff
@@ -273,198 +298,278 @@ describe('text.accessibleText', function() {
+		fixture.innerHTML = '<div id="t2label">This is <input type="text" value="the value" ' +
+			'aria-labelledby="t1label" aria-label="ARIA Label" id="t1"> of <i>everything</i></div>' +
+			'<div id="shadow"></div>';
+
+		var shadow = document.getElementById('shadow').attachShadow({ mode: 'open' });
+		shadow.innerHTML = '<div id="t1label">This is a <b>label</b></div>' +
+			'<label for="t1">HTML Label</label>' +
+			'<input type="text" id="t2" aria-labelledby="t2label">';
+
+		axe._tree = axe.utils.getFlattenedTree(fixture);
+		var target = axe.utils.querySelectorAll(axe._tree, '#t1')[0];
+		assert.equal(axe.commons.text.accessibleText(target), 'ARIA Label');
+	});
+
 	describe('figure', function() {
 
-		it('shoud check aria-labelledby', function() {
+		it('should check aria-labelledby', function() {
 			fixture.innerHTML = '<div id="t1">Hello</div>' +
 				'<figure aria-labelledby="t1">Not part of a11yName <figcaption>Fail</figcaption></figure>';
-			var target = fixture.querySelector('figure');
+			axe._tree = axe.utils.getFlattenedTree(fixture);
+
+			var target = axe.utils.querySelectorAll(axe._tree, 'figure')[0];
 			assert.equal(axe.commons.text.accessibleText(target), 'Hello');
 		});
 
-		it('shoud check aria-label', function() {
+		it('should check aria-label', function() {
 			fixture.innerHTML = '<figure aria-label="Hello">Not part of a11yName <figcaption>Fail</figcaption></figure>';
-			var target = fixture.querySelector('figure');
+			axe._tree = axe.utils.getFlattenedTree(fixture);
+
+			var target = axe.utils.querySelectorAll(axe._tree, 'figure')[0];
 			assert.equal(axe.commons.text.accessibleText(target), 'Hello');
 		});
 
-		it('shoud check the figures figcaption', function() {
+		it('should check the figures figcaption', function() {
 			fixture.innerHTML = '<figure>Not part of a11yName <figcaption>Hello</figcaption></figure>';
-			var target = fixture.querySelector('figure');
+			axe._tree = axe.utils.getFlattenedTree(fixture);
+
+			var target = axe.utils.querySelectorAll(axe._tree, 'figure')[0];
 			assert.equal(axe.commons.text.accessibleText(target), 'Hello');
 		});
 
-		it('shoud check title on figure', function() {
+		it('should check title on figure', function() {
 			fixture.innerHTML = '<figure title="Hello">Not part of a11yName <figcaption></figcaption></figure>';
-			var target = fixture.querySelector('figure');
+			axe._tree = axe.utils.getFlattenedTree(fixture);
+
+			var target = axe.utils.querySelectorAll(axe._tree, 'figure')[0];
 			assert.equal(axe.commons.text.accessibleText(target), 'Hello');
 		});
 
 		it('should not visit innerText of figure', function() {
 			fixture.innerHTML = '<figure>Hello<figcaption></figcaption></figure>';
-			var target = fixture.querySelector('figure');
+			axe._tree = axe.utils.getFlattenedTree(fixture);
+
+			var target = axe.utils.querySelectorAll(axe._tree, 'figure')[0];
 			assert.equal(axe.commons.text.accessibleText(target), '');
 		});
+
```

Comment:
I think we should add a couple more tests here;

1. fallback content for v1
2. correct use of an attribute (e.g. title) within the slotted shadow DOM content,
3. correct use of an attribute (e.g. title) within the shadow DOM content